src/Websites/Websites/Cmdlets/DeploymentSlots/SetAzureWebAppSlot.cs

```diff
@@ -298,6 +306,7 @@ public override void ExecuteCmdlet()
                     CmdletHelpers.TryParseAppServicePlanMetadataFromResourceId(WebApp.ServerFarmId, out rg, out servicePlanName);
                     WebApp.AzureStoragePath = null; // the API to update site Object doesn't have the AzureStorage Path property
                     WebsitesClient.UpdateWebApp(ResourceGroupName, location, Name, Slot, servicePlanName, WebApp);
+                    WebsitesClient.AddCustomHostNames(ResourceGroupName, location, Name, WebApp.HostNames.ToArray(), Slot);
```

Comment:
HostNames is already of type string[] so we don't need the ToArray().

Reply:
HostName is of type IList. So Conversion is required.